zanata-war/src/main/java/org/zanata/ApplicationConfiguration.java

```diff
@@ -239,20 +239,26 @@ public String getRegisterPath() {
     public String getServerPath() {
         String configuredValue = databaseBackedConfig.getServerHost();
         if (configuredValue != null) {
-            serverPath = configuredValue;
+            return configuredValue;
+        } else if (defaultServerPath != null) {
+            return defaultServerPath;
+        } else {
+            createDefaultServerPath();
+            return defaultServerPath;
```

Comment:
Don't need the `else` clauses since every code path has a return statement. Doesn't really make the code any more or less maintainable though.